src/objdump.main.v3

```diff
@@ -59,5 +62,88 @@ def parseAndDump(engine: Engine, path: string) -> int {
 		if (f.imp == null) OUT.put1(" { %d bytes }", f.orig_bytecode.length);
 		OUT.outln();
 	}
+	for (i < module.heaptypes.length) {
+		var h = module.heaptypes[i];
+		OUT.put1("heaptypes[%d] = ", i);
+		h.render(OUT);
+		var fh = h.getFirstSuperType();
+		if (fh != null) fh.render(OUT);
+		OUT.outln();
+	}
+	for (i < module.abstypes.length) {
+		var a = module.abstypes[i];
+		OUT.put1("abstypes[%d] = ", i);
+		if (a.isAssignableTo(ValueType.BOTTOM)) {
+			OUT.put1("%q ", ValueType.BOTTOM.render);
+		}
+		if (a.isAssignableTo(ValueType.I32)) {
+			OUT.put1("%q ", ValueType.I32.render);
+		}
+		if (a.isAssignableTo(ValueType.I64)) {
+			OUT.put1("%q ", ValueType.I64.render);
+		}
+		if (a.isAssignableTo(ValueType.F32)) {
+			OUT.put1("%q ", ValueType.F32.render);
+		}
+		if (a.isAssignableTo(ValueType.F64)) {
+			OUT.put1("%q ", ValueType.F64.render);
+		}
+		if (a.isAssignableTo(ValueType.V128)) {
+			OUT.put1("%q ", ValueType.V128.render);
+		}
+		OUT.outln();
+	}
+	for (i < module.tables.length) {
+		var t = module.tables[i];
+		OUT.put1("tables[%d] default elem = ", i);
+		match (t.default_elem) {
+			I32 => OUT.puts("I32");
+			I64 => OUT.puts("I64");
+			F32 => OUT.puts("F32");
+			F64 => OUT.puts("F64");
+			V128 => OUT.puts("V128");
+			FuncRefNull => OUT.puts("FuncRefNull");
+			ExternRefNull => OUT.puts("ExternRefNull");
+			Global => OUT.puts("Global");
+			FuncRef => OUT.puts("FuncRef");
+			Const => OUT.puts("Const");
+			I31 => OUT.puts("I31");
+			Array => OUT.puts("Array");
+			FixedArray => OUT.puts("FixedArray");
+			Struct => OUT.puts("Struct");
+		}
+		OUT.outln();
+	}
+	for (i < module.tags.length) {
+		var t = module.tags[i];
+		OUT.put1("tags[%d] = ", i);
+		if (t.fields != null) {
+			for (value in t.fields) {
+				OUT.put1("value: %q", value.render);
+				OUT.outln();
+			}
+		}
+		OUT.outln();
+	}
+	for (i < module.exports.length) {
+		var e = module.exports[i];
+		OUT.put1("exports[%d] = ", i);
+		OUT.puts(e.0);
+		match (e.1) {
+			x: SigDecl => OUT.puts(": SigDecl");
```

Comment:
Here, only functions, tables, memories, globals, and tags should be exportable. We can use the short names, "func", "table", "memory", "global", "tag", plus their respective index. (The reason for that is that `FuncDecl` and so on are names internal to Wizard, while the objdump utility should speak in terms that are WebAssembly-universal.).

Reply:
I'm confused by this line:

> We can use the short names, "func", "table", "memory", "global", "tag", plus their respective index.
Where do we use the short names? Is it inside the puts or we can just write Func instead of FuncDecl after the colon to represent type? Also, I didn't see where we can access their index. I saw that there's an import_index param in input of class ImportInfo but nothing inside the class can be used..?

Reply:
Just inside the string, so that the output doesn't say "FuncDecl", etc.